js/ohm/interpreter-semantics.js

```diff
@@ -457,6 +457,48 @@ const createInterpreterSemantics = (partContext, systemContext) => {
             };
         },
 
+        ThereIsAPropertyConditional_withSpecifier: function(thereLiteral, isLiteral, aLiteral, propertyLiteral, propName, ofLiteral, objectSpecifier){
```

Comment:
Could we simplify the number of rules here by specifying a `Factor` as the ultimate target of the rule (for example `there is a <factor>`)? Since Factors should be anything that produces a value I think it should work for everything from object specifiers to variables to property values

Reply:
One reason not to do this is because it would introduce a potential ambiguity.

At the moment `factor`'s can be property values, not properties themselves. Of course you can introduce property (i.e. `propertyName` rule) as a case for `factor` but then how do you know whether you asking a true/false response is addressing the value of the property or whether the property exists. 

In my view `there is` grammar should refer to the existence of things in the system (since we don't really have or want variables, that just means objects and properties), whereas `factors` can deal with things that evaluate (which I think was the HyperTalk idea, or at least it can be our version of their design). 

This way when you say 

```
if there is a [X]
....
```
vs 

```
if x+2
```
or

```
if the "prop-name" of objec
```

it's very clear in the grammar and semantics/interpreter what is happening 

@darth-cheney thoughts?